dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/forge/DataUploadConfigurationForgeryFactory.kt

```diff
@@ -14,6 +14,7 @@ import fr.xgouchet.elmyr.ForgeryFactory
 internal class DataUploadConfigurationForgeryFactory : ForgeryFactory<DataUploadConfiguration> {
     override fun getForgery(forge: Forge): DataUploadConfiguration {
         val frequency: UploadFrequency = forge.getForgery()
-        return DataUploadConfiguration(frequency)
+        // we limit the size to avoid OOM errors inside our tests
+        return DataUploadConfiguration(frequency, forge.anInt(min = 1, max = 50))
```

Comment:
the `maxBatchesPerUpload` gives the size of the batches list in some of our tests. That number can go up to `Int.MAX` and it creates problems ...at least I had those on my end. I guess a rule of thumb would be to try to stick with low size numbers when using lists of objects in our tests.

Reply:
yes, but since we have a definition `HIGH = 100`, we need to set at least 100 as max to make sure things work for `HIGH` as well (although the allocation pattern would eventually be different in unit-tests compared to the on-device run, it is still better to align imo).